spec/domain/command_handlers/game_command_handler_spec.rb

```diff
@@ -116,231 +137,335 @@
+          SpawnPlacedEvent.new(id, bob, GameUnit.new(0, 0, GameUnit::DOWN)),
+          GoalPlacedEvent.new(id, Goal.new(0, 0, 1)),
+          RobotSpawnedEvent.new(id, bob, GameUnit.new(0, 0, GameUnit::DOWN)),
+          GameRoundStartedEvent.new(id, GameRound.new(1), {'bob' => []}, {'bob' => []}),
+          InstructionCardDealtEvent.new(id, bob, InstructionCard.u_turn(10)),
+          InstructionCardDealtEvent.new(id, bob, InstructionCard.u_turn(20)),
+          InstructionCardDealtEvent.new(id, bob, InstructionCard.u_turn(30)),
+          InstructionCardDealtEvent.new(id, bob, InstructionCard.u_turn(40)),
+          InstructionCardDealtEvent.new(id, bob, InstructionCard.u_turn(50)),
+          InstructionCardDealtEvent.new(id, bob, InstructionCard.u_turn(60)),
+          InstructionCardDealtEvent.new(id, bob, InstructionCard.rotate_left(70)),
+          InstructionCardDealtEvent.new(id, bob, InstructionCard.rotate_left(90)),
+          InstructionCardDealtEvent.new(id, bob, InstructionCard.rotate_left(110)),
+          RobotProgrammedEvent.new(id, bob, [
+            InstructionCard.u_turn(10),
+            InstructionCard.u_turn(20),
+            InstructionCard.u_turn(30),
+            InstructionCard.u_turn(40),
+            InstructionCard.u_turn(50)
+          ]),
+          AllRobotsProgrammedEvent.new(id)
+        )
+      end
 
-        it_behaves_like "an event publisher" do
-          let(:expected_events) do
-            [
-              RobotRotatedEvent.new(id, bob, GameUnit.new(0, 0, GameUnit::UP)),
-              RobotRotatedEvent.new(id, bob, GameUnit.new(0, 0, GameUnit::DOWN)),
-              RobotRotatedEvent.new(id, bob, GameUnit.new(0, 0, GameUnit::UP)),
-              RobotRotatedEvent.new(id, bob, GameUnit.new(0, 0, GameUnit::DOWN)),
-              RobotRotatedEvent.new(id, bob, GameUnit.new(0, 0, GameUnit::UP)),
-              GoalTouchedEvent.new(id, bob, Goal.new(0, 0, 1)),
-              PlayerWonGameEvent.new(id, bob),
-              SpawnReplacedEvent.new(id, bob, GameUnit.new(0, 0, GameUnit::UP)),
-              InstructionCardDiscardedEvent.new(id, InstructionCard.u_turn(10)),
-              InstructionCardDiscardedEvent.new(id, InstructionCard.u_turn(20)),
-              InstructionCardDiscardedEvent.new(id, InstructionCard.u_turn(30)),
-              InstructionCardDiscardedEvent.new(id, InstructionCard.u_turn(40)),
-              InstructionCardDiscardedEvent.new(id, InstructionCard.u_turn(50)),
-              InstructionCardDiscardedEvent.new(id, InstructionCard.u_turn(60)),
-              InstructionCardDiscardedEvent.new(id, InstructionCard.rotate_left(70)),
-              InstructionCardDiscardedEvent.new(id, InstructionCard.rotate_left(90)),
-              InstructionCardDiscardedEvent.new(id, InstructionCard.rotate_left(110)),
-              GameEndedEvent.new(id, GameState::ENDED)
-            ]
-          end
-        end
+      it 'wins the game for a player' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.